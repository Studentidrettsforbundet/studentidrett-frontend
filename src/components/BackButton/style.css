.button{
    background-color: inherit;
    color: inherit;
    align-content: inherit;
    padding: 0px;
    padding-left: 10px
}

.button:hover {
    background-color: inherit;
    border: inherit;
    color: inherit;
    align-content: inherit;
    padding: 0px;
    padding-left: 10px
}