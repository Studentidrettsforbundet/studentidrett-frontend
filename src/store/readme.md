# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
