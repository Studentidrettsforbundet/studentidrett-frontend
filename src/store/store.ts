import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
