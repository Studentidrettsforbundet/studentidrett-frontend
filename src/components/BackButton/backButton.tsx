import React from 'react'
import { useHistory } from 'react-router-dom'
import { Button } from 'react-bootstrap';
import './style.css'

const BackButton = () => {
    const history = useHistory();
    const buttonClick = () => {
        history.go(-1);
    }
    return (<Button className='button' onClick={buttonClick}> Tilbake </Button>)

}

export default BackButton;