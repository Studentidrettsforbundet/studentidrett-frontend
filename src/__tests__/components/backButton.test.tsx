import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../../store/store';

import BackButton from '../../components/BackButton/backButton';

describe('backButton', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <BackButton />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
