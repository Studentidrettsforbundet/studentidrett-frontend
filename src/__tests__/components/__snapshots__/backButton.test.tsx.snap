// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backButton renders correctly 1`] = `
<button
  className="button btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  ‚Üê Tilbake
   
</button>
`;
